bin/ams-create-users-gocdb.py

```diff
@@ -0,0 +1,152 @@
+            continue
+
+        # Create AMS user
+        hostname = service_endpoint.find("HOSTNAME").text.replace(".", "-")
+        sitename = service_endpoint.find("SITENAME").text.replace(".", "-")
+        user_binding_name = service_type + "---" + hostname + "---" + sitename
+        print(hostname)
+        # reverse the dn and exclude the last slash
+        service_dn = ",".join(x for x in service_dn.text.split("/")[::-1][:-1])
+        project = {'project': ams_project, 'roles': [users_role]}
+        usr_create = {'projects': [project], 'email': ams_email}
+        print(json.dumps(usr_create))
+
+        # create the user
+        req = requests.post("https://"+ams_host+"/v1/users/" + user_binding_name + "?key=" + ams_token, data=json.dumps(usr_create), verify=verify)
+        print(req.text)
+        # if the rsponse doesn't contain the field uuid, it means the user was not created
+        req_data = json.loads(req.text)
+        if req.status_code != 201:
+            logger.log(logging.CRITICAL, "Something went wrong while creating ams user.\nBody data: " + str(usr_create) + "\nResponse Body: " + req.text)
+        if "uuid" not in req_data:
+            logger.log(logging.CRITICAL, "uuid field not found in response from ams")
+            continue
+
+        # Create the respective AUTH binding
+        bd_data = {'name': user_binding_name, 'service_uuid': authn_service_uuid, 'host': authn_service_host, 'dn': service_dn, 'unique_key': req_data["uuid"]}
+        rr = requests.post("https://"+authn_host+"/v1/bindings?key="+authn_token, data=json.dumps(bd_data), verify=verify)
+        print(rr.text)
+        if rr.status_code != 201:
+            logger.log(logging.CRITICAL, "Something went wrong while creating a binding.\nBody data: " + str(bd_data) + "\nResponse: " + rr.text)
+            # delete the ams user, since binding could not be created
+            requests.delete("https://"+ams_host+"/v1/users/" + user_binding_name + "?key=" + ams_token, verify=verify)
+            continue
+
+        # if both the user and binding have been created, assign the user to the acl of the topic
+        services[service_type].append(hostname)
+
+
+# modify the acl for each topic , to add all associated users
+    print(services)
+    for service in services:
+        # if there are no users in the acl, skip the next topic
+        if len(services[service]) == 0:
+            continue
+        print(service)
+        authorized_users = services[service]
+        rr = requests.post("https://"+ams_host+"/v1/projects/"+ams_project+"/topics/"+service+":modifyAcl?key="+ams_token, data=json.dumps({'authorized_users': authorized_users}), verify=verify)
+        # print(services[service])
+        # try:
+        #     ams.modifyacl_topic(topic=service, users=authorized_users, verify=False)
+        # except AmsException as e:
+        #     logger.log(logging.CRITICAL, e.msg)
+
+    print(missing_dns)
```

Comment:
Add a Description  "Missing DNs:"
Also a short summary of how many users were created would be useful

Reply:
ready